packages/server/graphql/mutations/addMissingJiraField.ts

```diff
@@ -101,10 +102,58 @@ const addMissingJiraField = {
         dimensionField.cloudId === cloudId &&
         dimensionField.projectKey === projectKey
     )
-    //TODO: Figure out how to add selected field to a proper Jira screen
-    // const project = await manager.getProject(cloudId, dimensionField.projectKey)
-    // const jiraScreens = manager.getAllScreens(cloudId)
-    await manager.addFieldToDefaultScreen(cloudId, dimensionField.fieldId)
+
+    const {values: screens} = (await manager.getScreens(cloudId)) as JiraScreensResponse
+
+    // we're trying to guess what's the probability that given screen is assigned to an issue project
+    const evaluateProbability = (screen: JiraScreen) => {
+      if (screen.name.startsWith(projectKey) && screen.name.includes('Default')) return 1
+      if (screen.name.includes(projectKey)) return 0.9
+      if (screen.name.includes('Bug')) return 0
+
+      return 0.5
+    }
+    const possibleScreens = screens
+      .map((screen) => ({screen, probability: evaluateProbability(screen)}))
+      .sort((screen1, screen2) => screen2.probability - screen1.probability)
+      .map(({screen}) => screen)
+
+    // stores screen tabs so we won't need to refetch it when reverting from previous operation
+    const screenTabIds = new Map()
+    // iterate over all the screens sorted by probability, try to update the given field
+    for (let i = 0; i < possibleScreens.length; i++) {
+      try {
+        const currentScreen = possibleScreens[i]
+        const previousScreen = i > 0 ? possibleScreens[i - 1] : null
+        if (previousScreen) {
```

Comment:
we were getting timeouts on this part, so let's run some stuff in parallel. we could also do some batch work (e.g. grab all the tabIds at once), up to you!

```ts
const addFieldToScreen = async () => {
  await manager.getScreenTabs(...)
  await manager.addFieldToScreenTab(...)
}  

await Promise.all([
  previousScreen && manage.removeFieldFromScreenTab(...),
  addFieldToScreen
])
```

Reply:
@mattkrick I've pushed some changes: https://github.com/ParabolInc/parabol/pull/4880/commits/d5376c0258bd9c314eb0b9b9340728166305842c and now it's not failing even in the worst case scenario. The most probable scenario is a bit slower now because I'm fetching all the screen ids now at the beginning. I think it's a good compromise though. I'd appreciate if you could take a look, thanks!